Gems/Terrain/Code/Source/Components/TerrainPhysicsColliderComponent.h

```diff
@@ -124,5 +124,6 @@ namespace Terrain
 
     private:
         TerrainPhysicsColliderConfig m_configuration;
+        bool m_terrainDataActive = true;
```

Comment:
Is it correct to initialize this as `true` instead of being `false` until `OnTerrainDataCreateEnd()` is called?

Reply:
+1

Reply:
Yeah I was thinking about a case when the component is created when there's an active terrain, but turns out there's an EBus connect policy to check for the existing handlers of TerrainDataRequestBus.
Hence we can set it to false indeed. 
Had to update a few unit tests to activate an entity after all mock handlers are configured